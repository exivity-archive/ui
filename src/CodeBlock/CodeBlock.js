import React from 'react'
import PropTypes from 'prop-types'
import { Box } from 'reakit'
import SyntaxHighlighter, {
  registerLanguage
} from 'react-syntax-highlighter/prism-light'
import jsx from 'react-syntax-highlighter/languages/prism/jsx'
import style from 'react-syntax-highlighter/styles/prism/solarizedlight'

import theme from '../theme'

registerLanguage('jsx', jsx)

const CodeBlock = ({ children, language }) => {
  // Fix font-family on code element
  const myStyle = {
    ...style,
    'code[class*="language-"]': {
      ...style['code[class*="language-"]'],
      fontFamily: theme.type.fonts.monospace.family
    }
  }

  return <Box as={SyntaxHighlighter}
    language={language}
    style={myStyle}
    customStyle={{
      fontFamily: theme.type.fonts.monospace.family,
      backgroundColor: theme.palette.marker
    }}>
    {children}
  </Box>
}

CodeBlock.propTypes = {
  children: PropTypes.node.isRequired,
  language: PropTypes.oneOf(['jsx']).isRequired
}

CodeBlock.defaultProps = {
  language: 'jsx'
}

export default CodeBlock
