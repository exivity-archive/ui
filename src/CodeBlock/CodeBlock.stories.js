import React from 'react'
import { storiesOf } from '@storybook/react'

import CodeBlock from './CodeBlock'

storiesOf('atoms|CodeBlock', module)
  .add('default', () => <CodeBlock>const Component = () => "Hi!"</CodeBlock>)
