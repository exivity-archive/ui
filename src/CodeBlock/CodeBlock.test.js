/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'

import CodeBlock from './CodeBlock'

test('renders basic CodeBlock', () => {
  const component = renderer.create(<CodeBlock />)
  expect(component.toJSON()).toMatchSnapshot()
})
