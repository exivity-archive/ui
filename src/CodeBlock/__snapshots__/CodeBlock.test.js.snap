// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders basic CodeBlock 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

<pre
  className="c0"
  style={
    Object {
      "MozHyphens": "none",
      "MozTabSize": "4",
      "OTabSize": "4",
      "WebkitHyphens": "none",
      "backgroundColor": "#fdf6e3",
      "borderRadius": "0.3em",
      "color": "#657b83",
      "fontFamily": "'Fira Mono', monospace",
      "hyphens": "none",
      "lineHeight": "1.5",
      "margin": undefined,
      "msHyphens": "none",
      "overflow": "auto",
      "padding": "1em",
      "tabSize": "4",
      "textAlign": "left",
      "whiteSpace": "pre",
      "wordBreak": "normal",
      "wordSpacing": "normal",
      "wordWrap": "normal",
    }
  }
>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "4",
        "OTabSize": "4",
        "WebkitHyphens": "none",
        "color": "#657b83",
        "fontFamily": "'Fira Mono', monospace",
        "hyphens": "none",
        "lineHeight": "1.5",
        "msHyphens": "none",
        "tabSize": "4",
        "textAlign": "left",
        "whiteSpace": "pre",
        "wordBreak": "normal",
        "wordSpacing": "normal",
        "wordWrap": "normal",
      }
    }
  />
</pre>
`;
